### PR TITLE
[#109] Part I: Improve unit tests for SSHKit

### LINKED ISSUE #109: Improve unit tests for `SSHKit` and `Context`
The unit tests for `SSHKit` and `SSHKit.Context` have room for improvement and should be more extensive.

### PR DESCRIPTION
**Pull request by pmeinhardt**
**Fixes #109 (part 1). Adds tests for `SSHKit` functions.**

For your review, please go through commit by commit.
It seemed impractical to open a separate PR for each of these.

Each individual commit should be small and focused enough to be easy to grok.
If you don't like some of them it's simple enough to kick out separate commits.

---

* [ ] Documented the change if necessary
* [x] Tested this PRs change (with unit and/or functional tests)
* [ ] Added this PRs change to CHANGELOG.md (in the `#master` section) if  necessary.

### CODE REVIEW

File: lib/sshkit.ex [hunk 1]
```diff
@@ -131,15 +132,16 @@ defmodule SSHKit do
 
   ```
   hosts = ["10.0.0.1", "10.0.0.2"]
-  options = [user: "admin", port: "2222"]
+  options = [user: "admin", port: 2222]
```
Comment (klappradla) [inline lib/sshkit.ex]: 👍

File: test/sshkit/sshkit_test.exs [hunk 2]
```diff
@@ -6,80 +6,153 @@ defmodule SSHKitTest do
...
+        %Host{name: "10.0.0.1", options: [port: 2222, user: "me"]},
+        %Host{name: "10.0.0.2", options: [port: 2222]}
       ]
-      context = SSHKit.context(hosts, @options)
-      assert context == @context_merged_options
+
+      assert context == %Context{hosts: hosts}
     end
 
     test "does not override host options with shared options" do
```
Comment (klappradla) [inline test/sshkit/sshkit_test.exs]: Maybe speak about "default options" here instead - now that the naming changed 🤔

Comment (pmeinhardt) [reply test/sshkit/sshkit_test.exs]: Makes perfect sense!

As a side note: I wasn't 100% sure about the rename, but – to me – "default" makes it clearer that individual host options override these when comparing to "shared options".

Comment (klappradla) [reply test/sshkit/sshkit_test.exs]: Yep, fully agree 👍 

As said, I actually didn't fully understand the original implementation 🙈

**Comment by pmeinhardt** [comment]
Awesome! Thanks a lot for your review @klappradla 🎸

**Pull request merged**